jf_agent/git/bitbucket_cloud_adapter.py

```diff
@@ -161,35 +161,43 @@ def get_repos(
 
     @diagnostics.capture_timing()
     @agent_logging.log_entry_exit(logger)
-    def get_default_branch_commits(
-        self, normalized_repos: List[NormalizedRepository], server_git_instance_info,
+    def get_commits_for_included_branches(
+        self, normalized_repos: List[NormalizedRepository], included_branches: dict, server_git_instance_info,
     ) -> List[NormalizedCommit]:
-        print('downloading bitbucket default branch commits... ', end='', flush=True)
+        print('downloading bitbucket commits on included branches... ', end='', flush=True)
         for i, repo in enumerate(normalized_repos, start=1):
             with agent_logging.log_loop_iters(logger, 'repo for branch commits', i, 1):
                 pull_since = pull_since_date_for_repo(
                     server_git_instance_info, repo.project.login, repo.id, 'commits'
                 )
-                for j, commit in enumerate(
-                    tqdm(
-                        self.client.get_commits(repo.project.id, repo.id, repo.default_branch_name),
-                        desc=f'downloading commits for {repo.name}',
-                        unit='commits',
-                    ),
-                    start=1,
-                ):
-                    with agent_logging.log_loop_iters(logger, 'branch commit inside repo', j, 100):
-                        commit = _normalize_commit(
-                            commit,
-                            repo,
-                            self.config.git_strip_text_content,
-                            self.config.git_redact_names_and_urls,
-                        )
-                        yield commit
 
-                        # yield one commit older than we want to see
-                        if commit.commit_date < pull_since:
-                            break
+                # Find branches for which we should pull commits, specified by customer in config.
+                # If specific branches are not specified, just pull from default branch.
+                branches_for_repo = included_branches.get(repo.name)
+                branches = branches_for_repo if branches_for_repo else [repo.default_branch_name]
```

Comment:
I think the behavior we want is to always pull the default branch for all repos, and not force customers to specify the default branch if they specify extra branches

Reply:
True this may be misleading. New version always adds the default branch for a repo into the list we iterate over when fetching commits.